mysql-test/suite/innodb/t/ts_encrypt.test

```diff
@@ -0,0 +1,174 @@
+# InnoDB general tablespace encryption
+
+call mtr.add_suppression("Partition.*does not exist in the InnoDB internal data dictionary though");
+
+--source include/have_innodb.inc
+--source include/not_embedded.inc
+
+let $innodb_file_per_table = `SELECT @@innodb_file_per_table`;
+let $innodb_strict_mode = `SELECT @@innodb_strict_mode`;
+
+CREATE TABLESPACE ts_encrypted ADD DATAFILE 'ts_encrypted.ibd' ENCRYPTION="Y" ENGINE="InnoDB";
+CREATE TABLESPACE ts_unencrypted ADD DATAFILE 'ts_unencrypted.ibd' ENGINE="InnoDB";
+
+--error ER_ILLEGAL_HA_CREATE_OPTION
+CREATE TABLE t1 (a TEXT) TABLESPACE ts_encrypted ENGINE="InnoDB";
+
+CREATE TABLE t2 (a TEXT) TABLESPACE ts_unencrypted ENGINE="InnoDB";
+INSERT INTO t2 (a) VALUES ('ONE'), ('TWO'), ('THREE');
+
+CREATE TABLE t3 (a TEXT) TABLESPACE ts_encrypted ENCRYPTION="Y" ENGINE="InnoDB";
+INSERT INTO t3 (a) VALUES ('ONE'), ('TWO'), ('THREE');
+
+--error ER_ILLEGAL_HA_CREATE_OPTION
+CREATE TABLE t4 (a TEXT) TABLESPACE ts_unencrypted ENCRYPTION="Y" ENGINE="InnoDB";
+
+--error ER_ILLEGAL_HA_CREATE_OPTION
+CREATE TABLE t5 (a TEXT) TABLESPACE ts_encrypted ENCRYPTION="N" ENGINE="InnoDB";
+
+CREATE TABLE t6 (a TEXT) TABLESPACE ts_unencrypted ENCRYPTION="N" ENGINE="InnoDB";
+INSERT INTO t6 (a) VALUES ('ONE'), ('TWO'), ('THREE');
+
+--error ER_ILLEGAL_HA_CREATE_OPTION
+ALTER TABLE t2 ENCRYPTION='Y';
+--error ER_ILLEGAL_HA_CREATE_OPTION
+ALTER TABLE t3 ENCRYPTION='N';
+--error ER_ILLEGAL_HA_CREATE_OPTION
+ALTER TABLE t6 ENCRYPTION='Y';
+
+
+--error ER_ILLEGAL_HA_CREATE_OPTION
+ALTER TABLE t2 TABLESPACE ts_encrypted;
+--error ER_ILLEGAL_HA_CREATE_OPTION
+ALTER TABLE t3 TABLESPACE ts_unencrypted;
+--error ER_ILLEGAL_HA_CREATE_OPTION
+ALTER TABLE t6 TABLESPACE ts_encrypted;
+
+CREATE TABLESPACE ts_encrypted_new ADD DATAFILE 'ts_encrypted_new.ibd' ENCRYPTION="Y" ENGINE="InnoDB";
```

Comment:
Please, also check adding second datafile to existing  encrypted / unencrypted tablespace specifying `ENCRYPTION="Y"/"N"`